packages/solid/web/src/index.ts

```diff
@@ -99,9 +100,8 @@ export function Portal<T extends boolean = false, S extends boolean = false>(pro
   return marker;
 }
 
-type DynamicProps<T> = T & {
-  children?: any;
-  component?: Component<T> | string | keyof JSX.IntrinsicElements;
+type DynamicProps<T extends ValidComponent> = ComponentProps<T> & {
+  component?: T;
```

Comment:
As mentioned on discord I would like to know why `component` is optional. If the intent is to allow `undefined` to unmount the dynamic component then `T | undefined` would be better as omitting `component` completely doesn't seem very useful.

Reply:
@ryansolid

Reply:
Yeah there is no reason to omit it. In terms of unmounting it, I'm pretty sure wasn't my original intent but someone probably suggested it as something they might do. It probably makes sense to allow using it as a conditional I suppose. But yeah I think the `component` prop should be present even if it is allowed to be undefined.

Reply:
I have applied the change, thank you